cli/src/routes/validation-route.ts

```
@@ -0,0 +1,70 @@
+
+    private schemaDirectoryPath: string;
+    private schemaDirectory: SchemaDirectory;
+
+    constructor(router: Router, schemaDirectoryPath: string) {
+        this.schemaDirectory = new SchemaDirectory(true)
+        this.schemaDirectoryPath = schemaDirectoryPath
+        this.initializeRoutes(router);
+    }
+
+    private initializeRoutes(router: Router) {
+        router.post('/', this.validateSchema);
+        router.get('/health', this.healthCheck);
+    }
+
+    private validateSchema = async (req: Request, res: Response) => {
+        const schema = req.body['$schema'];
+        if (!schema) {
+            return res.status(400).json({ error: 'The "$schema" field is missing from the request body.' });
+        }
+        console.info("Path loading schemas is " + this.schemaDirectoryPath)
+
+        await this.schemaDirectory.loadSchemas(this.schemaDirectoryPath)
+        console.info("Loaded schemas: " + this.schemaDirectory.getLoadedSchemas())
+        const foundSchema = this.schemaDirectory.getSchema(schema)
+        if (foundSchema == undefined) {
+            return res.status(400).json({ error: 'The "$schema" field referenced is not available to the server' });
+        }
+        const tempInstantiation = await createTemporaryFile()
+        const tempPattern = await createTemporaryFile()
+        try {
+
+            await fs.writeFile(tempInstantiation, JSON.stringify(req.body, null, 4));
+            await fs.writeFile(tempPattern, JSON.stringify(foundSchema, null, 4));
+
+            const outcome = await validate(tempInstantiation, tempPattern, this.schemaDirectoryPath, true);
+            const content = getFormattedOutput(outcome, 'json');
+            res.status(201).type('json').send(content);
+        } catch (error) {
+            res.status(500).json({ error: error.message });
+        } finally {
+            [tempInstantiation, tempPattern].forEach(element => {
+                fs.unlink(element).catch(() => { });
```

What does this do? is this the node/fs equivalent of running 'rm'?

---

Also I see we don't do anything if they fail - basically swallowing the exception. I get that we don't want a failure in cleanup to return a bad status code, but we should at least log. Otherwise the host could fill up with temp files